### ISSUE TITLE
--no-default-features build broken

### ISSUE THREAD

**Issue by michaelkirk** [pka/http-range-client#3]
`reqwest` was made "optional" in https://github.com/pka/http-range-client/commit/37e04563ac2d7fcfae56fdeaf0445fc5ad3954e2, but it's not actually possible to build without that feature enabled.

```
$ cargo test --no-default-features
   Compiling http-range-client v0.6.0 (/Users/mkirk/src/georust/http-range-client)
error[E0432]: unresolved import `crate::HttpClient`
 --> src/buffered_range_client.rs:2:5
  |
2 | use crate::HttpClient;
  |     ^^^^^^^^^^^^^^^^^ no `HttpClient` in the root
```

**Comment by pka** [comment]
The idea was that you can choose which HTTP library to use. This makes the features non-additive (which is not good) and you have to set exactly one features. Better ideas welcome...

**Comment by michaelkirk** [comment]
Ah, OK. I can see why you'd want reqwest to be optional. Given that there's not really any other option than to use reqwest at this point, it seems like unnecessary confusion with no benefit. It's not a big deal either way though.

**Comment by kylebarron** [comment]
At some point I may make a PR to add an [`ehttp`](https://github.com/emilk/ehttp)-based client, because it looks to be a nice minimal library for native + wasm support. I'm curious whether reqwest would have a bigger bundle size increase

**Comment by pka** [comment]
ehttp looks interesting!

A big enhancement for http-range-client would be implementing `Read+Seek`. I think this shouldn't be very difficult and it would make it compatible with most file readers. But this requires switching to a sync API. So maybe we could introduce an additional sync API when including ehttp?

The use case for `Read+Seek` would be adding http support to libs like https://github.com/pka/copc-rs and https://github.com/pka/georaster

**Comment by michaelkirk** [comment]
>  it looks to be a nice minimal library for native + wasm support

Maybe ehttp is a better option, I don't know much about it, but just in case there's confusion: reqwest (and rust flatgeobuf) already work on wasm32. We used it here: https://github.com/a-b-street/abstreet/blob/main/popdat/src/import_census.rs#L27

**Comment by kylebarron** [comment]
I saw the support included, though I wasn't able to get it to work. I had to set `default-features = false` on `flatgeobuf` when building for `wasm32` or I got the below errors. I haven't tried to debug/see what's going on yet. When I make a small repro example I'll create a new issue.

```
error: future cannot be sent between threads safely
  --> /Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-range-client-0.6.0/src/reqwest_client.rs:11:72
   |
11 |       async fn get_range(&self, url: &str, range: &str) -> Result<Bytes> {
   |  ________________________________________________________________________^
12 | |         let response = self
13 | |             .get(url)
14 | |             .header("Range", range)
...  |
24 | |             .map_err(|e| HttpError::HttpError(e.to_string()))
25 | |     }
   | |_____^ future created by async block is not `Send`
   |
   = help: within `[async block@/Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-range-client-0.6.0/src/reqwest_client.rs:11:72: 25:6]`, the trait `Send` is not implemented for `*mut u8`
note: future is not `Send` as this value is used across an await
  --> /Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-range-client-0.6.0/src/reqwest_client.rs:23:14
   |
12 |         let response = self
   |             -------- has type `Response` which is not `Send`
...
23 |             .await
   |              ^^^^^ await occurs here, with `response` maybe used later
24 |             .map_err(|e| HttpError::HttpError(e.to_string()))
25 |     }
   |     - `response` is later dropped here
   = note: required for the cast from `Pin<Box<...>>` to `Pin<Box<dyn Future<Output = ...> + Send>>`
   = note: the full name for the source type has been written to '/Users/kyle/github/rust/geoarrow-rs/js/target/wasm32-unknown-unknown/release/deps/http_range_client-97c77ef0bcc290cc.long-type-13351506641901377478.txt'
   = note: the full name for the target type has been written to '/Users/kyle/github/rust/geoarrow-rs/js/target/wasm32-unknown-unknown/release/deps/http_range_client-97c77ef0bcc290cc.long-type-1228816590379566913.txt'
```

**Comment by pka** [comment]
This did work once: https://github.com/pka/flatgeobuf-wasm

**Comment by michaelkirk** [comment]
It's still working in abstreet today, see the [`popdat`](https://github.com/a-b-street/abstreet/tree/main/popdat) crate. I recall it required some target-arch specific feature configuration though.

**Comment by pka** [comment]
Fixed with #5
